doc/sources/.static/kivy.js

```diff
@@ -233,4 +233,19 @@ $(function() {
     			return this.nodeType === 3; //https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType
 		}).remove();
   });
-});
+
+	// to have the current "active" element be always be centred on the left side bar
+	const currApiIndex_li = $(".api-index li.current");
+	if (currApiIndex_li.length){
+		const target = currApiIndex_li[0];
+		$(target.parentNode).animate({"scrollTop": target.offsetTop - 2*target.parentNode.offsetTop }, 'normal');
+	}
+
+	document.documentElement.className = localStorage.getItem('theme') || "light";
```

Comment:
using `(prefers-color-scheme` and `matchMedia` ?

Reply:
Yes, and that line, that doesn't uses jQuery could also be placed into the <head>, in order to avoid that at refresh, the page doesn't flash between the two themes.

Reply:
Adding:
```html
<script>
     const currTheme = localStorage.getItem("theme");
     if(currTheme){
        document.documentElement.className = currTheme;
     } else if(window.matchMedia){
        document.documentElement.className = window.matchMedia('(prefers-color-scheme: dark)').matches ? "dark": "light";
     } else { // neither localStorage or window.matchMedia exists
        document.documentElement.className = "light"; //todo should dark theme be the fallback value instead ?
     }
</script>
````

just after the line `122` of `layout.html` and deleting the related part in `js` code should fix the issue.

Reply:
what if we add it above line `97` ?

Reply:
Better to keep it inside the `{%- block scripts %}` block

Reply:
i did but the code in not being included into the final rendered page

Reply:
nvm it was supposed to be on LN `124`